scripts/src/java/org/oppia/android/scripts/coverage/CoverageRunner.kt

```diff
@@ -36,23 +36,23 @@ class CoverageRunner(
    */
   fun retrieveCoverageDataForTestTarget(
     bazelTestTarget: String
-  ): CoverageReport {
-    val coverageResult = retrieveCoverageResult(bazelTestTarget)
-      ?: return generateFailedCoverageReport(
-        bazelTestTarget,
-        "Coverage retrieval failed for the test target: $bazelTestTarget"
-      )
-
-    return coverageDataFileLines(coverageResult, bazelTestTarget)
-  }
+  ): List<CoverageReport> {
+    val coverageResults = bazelClient.runCoverageForTestTarget(bazelTestTarget)
 
-  private fun retrieveCoverageResult(
-    bazelTestTarget: String
-  ): List<String>? {
-    return bazelClient.runCoverageForTestTarget(bazelTestTarget)
+    return coverageResults
+      .takeIf { it.isNotEmpty() }
+      ?.map { singleCoverageDatFileLines ->
+        parseCoverageDataFileLines(singleCoverageDatFileLines, bazelTestTarget)
+      }
+      ?: listOf(
```

Comment:
```suggestion
      .map { singleCoverageDatFileLines ->
        parseCoverageDataFileLines(singleCoverageDatFileLines, bazelTestTarget)
      }.takeIf { it.isNotEmpty() } ?: listOf(
```

Maybe a bit easier to parse when reading this.

Reply:
Okay. updated as suggested.